### PR TITLE
Update ForceField to store XML internally, add very basic parameter moves

### PR DESCRIPTION
**Pull request by davidlmobley**
This will update `ForceField` to retain the XML internally and allow its modification (i.e. for parameter moves). The XML will still be parsed on `ForceField` initialization, as per #84 , but if it is modified, it will trigger re-parsing and storing of forces on application of `createSystem`. 

I expect I'll implement some basic functionality here for interacting with and manipulating parameters, such as:
- `GetParameter`, which would return info associated with a specific parameter as identified either by SMIRKS or parameter ID and force type
- `SetParameter`, mirroring `GetParameter` but for setting a particular parameter
- `GetParameterID`, returning just the parameter ID for a particular parameter
- `GetParameterIDs`, getting all parameter IDs for a particular section (needed for determining unique parameter IDs for new child parameters)

Presumably later we will also need the ability to introduce new parameters, such as something like:
- `CreateChildParameter(smirks, parentsmirks, section, paramdict)`, 
- `DeleteChildParameter` which would delete a specified parameter which is a child of another parameter

Along with that latter set, we would probably also need something which tracks the parameter hierarchy so we know which parameters can be deleted (at least without substantial work, only parameters which have no children can be deleted).

### CODE REVIEW

**Comment by jchodera** [comment]
Looks good to me.

**Comment by davidlmobley** [comment]
Updated to remove unused code which attempted to handle `AtomTypes` (which we're not using) as per #85 .

**Comment by davidlmobley** [comment]
Added `getParameter`, `setParameter`, and `writeFile`.

**Comment by davidlmobley** [comment]
Updated to add a (detailed) test and an example of parameter manipulation in an IPython notebook under `examples/forcefield_modification`. 

Should be ready to merge as this has the basics (get a parameter, set a parameter, write XML). Other functionality will be discussed in a separate issue and added under another PR.

@bannanc - take a quick look and try my IPython notebook under `examples` if you don't mind...?

File: smarty/forcefield.py [hunk 1]
```diff
@@ -389,9 +386,144 @@ def registerGenerator(self, generator):
...
+
+Usage notes: SMIRKS or parameter ID must be specified.
+
+To do: Update behavior of "Implied" force_type so it raises an exception if the parameter is not uniquely identified by the provided info.
+"""
+        # Check for valid input
+        if smirks and paramID:
+            raise ValueError("Error: Specify SMIRKS OR parameter ID but not both.")
+        if smirks==None and paramID==None:
+            raise ValueError("Error: Must specify SMIRKS and/or parameter ID.")
```
Comment (bannanc) [inline smarty/forcefield.py]: You raise an error if both smirks and paramID are specified, doesn't that mean the Error message here should say "SMIRKS or parameter..." not "and/or"?

File: smarty/forcefield.py [hunk 2]
```diff
@@ -389,9 +386,144 @@ def registerGenerator(self, generator):
...
```
Comment (bannanc) [inline smarty/forcefield.py]: "and/or" --> "or"?

Comment (bannanc) [inline smarty/forcefield.py]: "and/or" to "or"?

**Comment by bannanc** [comment]
@davidlmobley it took me a little bit to try to follow this since I haven't been working on this side of the project. I made comments on the code lines, but there were a few Error messages that I think need to be or not and/or. That is assuming I understood your if statements  correctly. 

Otherwise, I think I followed the ipython notebook and it looks good.

**Comment by davidlmobley** [comment]
Thanks, @bannanc .

**Pull request merged**